core/src/test/java/org/elasticsearch/recovery/RelocationIT.java

```diff
@@ -375,6 +368,9 @@ public void indexShardStateChanged(IndexShard indexShard, @Nullable IndexShardSt
                 }
             }
 
+            // refresh is a replication action so this forces a global checkpoint sync which is need as these are asserted on in tear down
+            client().admin().indices().prepareRefresh("test").get();
```

Comment:
I wonder if we should randomize this and make the assertion in `beforeIndexDeletion` on "global checkpoint mismatch" conditional on whether we refreshed or not. This allows us to check check the other assertions, even if there is no refresh (which could possibly update local checkpoint).